### ISSUE TITLE
Can't build depyt

### ISSUE THREAD

**Issue by phaer** [rizo/rekube#1]
Hi, thanks for the effort!

I tried to build this project, first as a dependency as described in the readme, then just this repo reproducing the following error:

```
error: build failed with exit code: 1
  build log:
    # esy-build-package: building: @opam/depyt@github:rizo/depyt:depyt.opam#ca2ccf8                                   
    # esy-build-package: pwd: /root/.esy/3/b/opam__s__depyt-60aa73b0                                                  
    # esy-build-package: running: 'dune' 'subst'
             git (internal) (exit 128)
    /usr/bin/git describe --always --dirty > /tmp/dune19a26a.output                                                   
    fatal: Not a git repository (or any of the parent directories): .git                                              
             git (internal) (exit 128)
    /usr/bin/git rev-parse HEAD > /tmp/dune5bee0d.output
    fatal: Not a git repository (or any of the parent directories): .git                                              
             git (internal) (exit 128)
    /usr/bin/git ls-tree -r --name-only HEAD > /tmp/dunee6e174.output                                                 
    fatal: Not a git repository (or any of the parent directories): .git                                              
    error: command failed: 'dune' 'subst' (exited with 1)
    esy-build-package: exiting with errors above...

  building @opam/depyt@github:rizo/depyt:depyt.opam#ca2ccf8    
```
I see you are using your own fork of depyt and I thought maybe that's related?

For reference and reproduction, I am using the following Dockerfile:

```
FROM node:latest
WORKDIR /root
RUN npm install --unsafe-perm=true -g esy
RUN git clone https://github.com/rizo/rekube
```

**Comment by rizo** [comment]
Hey! Thanks for reporting this.

I created a minimal `package.json` to try to reproduce this issue:

```json
{
  "name": "esy-depyt",
  "dependencies": {
    "@opam/depyt": "github:rizo/depyt:depyt.opam#ca2ccf8"
  },
  "devDependencies": {
    "ocaml": "4.6.x",
    "@opam/dune": "*"
  }
}
```

With this I'm getting the same problem:

```
error: build failed with exit code: 1
  build log:
    # esy-build-package: building: @opam/depyt@github:rizo/depyt:depyt.opam#ca2ccf8
    # esy-build-package: pwd: /root/.esy/3/b/opam__s__depyt-8c770136
    # esy-build-package: running: 'dune' 'subst'
             git (internal) (exit 128)
    /usr/bin/git describe --always --dirty > /tmp/dune951ae1.output
    fatal: Not a git repository (or any of the parent directories): .git
             git (internal) (exit 128)
    /usr/bin/git rev-parse HEAD > /tmp/dunea63bae.output
    fatal: Not a git repository (or any of the parent directories): .git
             git (internal) (exit 128)
    /usr/bin/git ls-tree -r --name-only HEAD > /tmp/dune75e5db.output
    fatal: Not a git repository (or any of the parent directories): .git
    error: command failed: 'dune' 'subst' (exited with 1)
    esy-build-package: exiting with errors above...
```

I checked `~/.esy/3/b/opam__s__depyt-8c770136` and for some reason its `.git` directory is empty.

@andreypopp, do you think it's worth reporting this as an esy issue?

I'll continue the investigation.

**Comment by rizo** [comment]
This problem seems to have been introduced in recent versions of esy. I opened an issue on esy's project: https://github.com/esy/esy/issues/1056.

---

The following Dockerfile with `easy@0.5` builds successfully:

```docker
FROM node:latest
WORKDIR /root
RUN npm install --unsafe-perm=true -g esy@0.5
RUN git clone https://github.com/rizo/rekube
WORKDIR /root/rekube
RUN esy
```

---

I added the above Dockerfile to rekube and added a Docker Hub CI. The image should be available shortly as `odis/rekube:latest` once the build finishes: https://hub.docker.com/repository/docker/odis/rekube/tags

**Comment by phaer** [comment]
I can confirm this works, thanks a lot for amazingly fast testing and this work-around :+1:

**Comment by rizo** [comment]
No worries, glad I could help.

**Comment by mthssdrbrg** [comment]
Hiya :wave:!

I seem to be running into another issue with building depyt (when trying to get the quickstart working):

```
error: build failed with exit code: 1
  build log:
    # esy-build-package: building: @opam/depyt@github:rizo/depyt:depyt.opam#ca2ccf8
    # esy-build-package: pwd: /root/.esy/3__________________________________________________________________________/b/opam__s__depyt-d2983ecc
    # esy-build-package: running: 'dune' 'subst'
    # esy-build-package: running: 'dune' 'build' '-p' 'depyt' '-j' '4'
          ocamlc src/.depyt.objs/byte/depyt.{cmi,cmo,cmt} (exit 2)
    (cd _build/default && /root/.esy/3__________________________________________________________________________/i/ocaml-4.6.1000-46ec027d/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.depyt.objs/byte -I /root/.esy/3__________________________________________________________________________/i/opam__s__bigarray_compat-opam__c__1.0.0-a370ebf5/lib/bigarray-compat -I /root/.esy/3__________________________________________________________________________/i/opam__s__cstruct-opam__c__5.1.1-0f1506c7/lib/cstruct -I /root/.esy/3__________________________________________________________________________/i/opam__s__fmt-opam__c__0.8.8-00b86735/lib/fmt -I /root/.esy/3__________________________________________________________________________/i/opam__s__jsonm-opam__c__1.0.1-b9d7b18d/lib/jsonm -I /root/.esy/3__________________________________________________________________________/i/opam__s__ocamlfind-opam__c__1.8.1-c7edfb4b/lib/bytes -I /root/.esy/3__________________________________________________________________________/i/opam__s__ocplib_endian-opam__c__1.1-612fed51/lib/ocplib-endian -I /root/.esy/3__________________________________________________________________________/i/opam__s__result-opam__c__1.5-7f5cebf3/lib/result -I /root/.esy/3__________________________________________________________________________/i/opam__s__seq-opam__c__0.2.2-b2785b28/lib/seq -I /root/.esy/3__________________________________________________________________________/i/opam__s__stdlib_shims-opam__c__0.1.0-6891926a/lib/stdlib-shims -I /root/.esy/3__________________________________________________________________________/i/opam__s__uchar-opam__c__0.0.2-07de315a/lib/uchar -I /root/.esy/3__________________________________________________________________________/i/opam__s__uutf-opam__c__1.0.2-57e1ce9f/lib/uutf -no-alias-deps -o src/.depyt.objs/byte/depyt.cmo -c -impl src/depyt.ml)
    File "src/depyt.ml", line 756, characters 15-37:
    Warning 3: deprecated: Cstruct.blit_to_string
    Use blit_to_bytes instead, blit_to_string will be removed in cstruct 5.0.0
    File "src/depyt.ml", line 1086, characters 21-24:
    Error: The type constructor t expects 2 argument(s),
           but is here applied to 1 argument(s)
    error: command failed: 'dune' 'build' '-p' 'depyt' '-j' '4' (exited with 1)
    esy-build-package: exiting with errors above...

  building @opam/depyt@github:rizo/depyt:depyt.opam#ca2ccf8
esy: exiting due to errors above
The command '/bin/sh -c esy build' returned a non-zero code: 1
```

Minimal Dockerfile for reproducing the issue:
```dockerfile
FROM node:14.0

RUN npm install --unsafe-perm=true -g esy@0.5

RUN mkdir /src && \
  printf '{\n  "name": "rekube-test",\n  "version": "0.1.0",\n  "dependencies": {\n    "rekube": "github:odis-labs/rekube#f9d8e20"\n  }\n}' \
  > /src/package.json && \
  cat /src/package.json

WORKDIR /src
RUN esy install --solve-timeout=300
RUN esy build
```

There's a PR upstream that attempts to fix the issue https://github.com/samoht/depyt/pull/21 (I think at least), but I'm not entirely sure how to go about integrating it in rekube. Any pointers / help would be greatly appreciated :)

Wasn't sure whether to create a new issue or comment on this, happy to create a new issue if you prefer!

**Comment by rizo** [comment]
Hi @mthssdrbrg! 😃 Thanks for reporting this.

It seems like there's an issue with Depyt and the latest versions of the OCaml compiler.

To resolve this issue I decided to completely remove the dependency on Depyt. Depyt was [being used](https://github.com/odis-labs/rekube/commit/11add43ec35456a20e7d30c593bee499f44365ce) by an experimental and incomplete tool called `rekube-conv` used to convert YAML files to Rekube files. I plan to add this tool back at some point in the future as an independent package. For now, this allows Rekube to focus on providing the Kubernetes API and the configuration DSL with minimal dependencies.

Your Dockerfile should now build successfully (don't forget to use 0ce7f9f).

I also created a [rekube-starter](https://github.com/odis-labs/rekube-starter) project with an example deployment. This should be an easier starting point, if you want to try it.
